### ISSUE TITLE
Allow configuration of email and SMS fields just like index/search/show fields

### ISSUE THREAD

**Issue by dkinzer** [projectblacklight/blacklight#1923]
In BL I can configure various field mapping for search, index or showing of a record via methods such as `blacklight_config.add_show_field`.

This may lead a user to suppose that there are equivalent ways of configuring fields for the SMS and email functionality.  But instead they will find out that there is a completely different way that fields for an email are configured.  In fact the configuration is not even as flexible.

For example, the current code that generates the body of a document email is:

```ruby
# frozen_string_literal: true
# This module provides the body of an email export based on the document's semantic values
module Blacklight::Document::Email
  # Return a text string that will be the body of the email
  def to_email_text
    semantics = to_semantic_values
    body = []
    body << I18n.t('blacklight.email.text.title', value: semantics[:title].join(" ")) unless semantics[:title].blank?
    body << I18n.t('blacklight.email.text.author', value: semantics[:author].join(" ")) unless semantics[:author].blank?
    body << I18n.t('blacklight.email.text.format', value: semantics[:format].join(" ")) unless semantics[:format].blank?
    body << I18n.t('blacklight.email.text.language', value: semantics[:language].join(" ")) unless semantics[:language].blank?
    return body.join("\n") unless body.empty?
  end
end
``` 

We can see from the get go that this is very different from how other fields are configured in BL.  The use of the `to_semantic_values` here is different from the use `blacklight_config.*` in other context.  Further the specific semantic fields that get used in the email are hard_coded.

As far as I can see there is no way to configure these fields outside of overriding the `semantic_fields` method and no way to update the body email but to override this `to_email_text` method.

I think it should be possible to update this method to use a blacklight configuration with the same API usage as that of `blacklight_config.add_search_field` etc....  This would make it consistent with the rest of the BL project.

**Comment by barmintor** [comment]
Are there backwards compatibility issues in #1988 that require scoping to 7.0 or 8.0, or can this be addressed in a feature release?

**Comment by dkinzer** [comment]
@barmintor no, there are no backwards compatible issues as I made sure to keep backwards compatibility with the current way of configuring these fields using the semantics fields technique.

**Comment by dkinzer** [comment]
Closed by #1988
